### PR TITLE
CMake Presets + Readme Update

### PR DESCRIPTION
**Pull request by McCallisterRomer**
Updated the readme build instructions and added CMake presets for building in Visual Studio.

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -6,54 +6,122 @@
 
 NcEngine is a 3D game engine written in C++20 targeting Windows. It is actively being developed but is still in very early stages. Features are in varying stages of completeness.
 
-## Links
 -----------
+Getting started:
```
Comment (nightcrawlerstudios) [inline readme.md:10]: Should be the same level of header as Requirements.

File: readme.md [hunk 2]
```diff
@@ -6,54 +6,122 @@
...
------------
-Current limitations:
-* NcEngine depends on Windows. There will likely be cross-platform support in the future, but it is not actively being pursued.
-* There are a few places where gcc specific code hasn't been fixed. This unfortunately means that mingw is the only currently supported compiler. This is planned to be fixed in the near future.
-
-There are three CMake projects in the repository
-* The engine, located in the 'nc' directory, is built as a static library and is output to 'nc/lib'.
-* An editor, located in the 'editor' directory, is output to the repository directory as 'NcEditor.exe'.
-* A sample project, located in the 'project' directory, is output to the repository directory as 'Sample.exe'.
+### Requirements:
```
Comment (nightcrawlerstudios) [inline readme.md:25]: The ':" are inconsistent in the headers - I don't have a preference for ":" or none however, either way.

File: readme.md (hunk 1 above)
Comment (nightcrawlerstudios) [reply readme.md:10]: I would also like to see the ------- line break underneath Getting Started

File: readme.md [hunk 3]
```diff
@@ -6,54 +6,122 @@
 
 NcEngine is a 3D game engine written in C++20 targeting Windows. It is actively being developed but is still in very early stages. Features are in varying stages of completeness.
 
-## Links
 -----------
+Getting started:
+* [Requirements](#requirements)
+* [Targets](#targets)
 * [Building](#building)
```
Comment (nightcrawlerstudios) [inline readme.md:13]: Targets should go inside of building IMO, and be a smaller header

File: readme.md [hunk 4]
```diff
@@ -6,67 +6,121 @@
...
+* WithProfiling: Include Optick profiling code.
+* WithValidation: Enable Vulkan validation layers.
+
+Each preset provides a different set of [definitions](#definitions). When adding custom presets to CMakeUserPresets.json, NC_INSTALL_DIR must be defined.
+
+More on CMake Presets:
+* [Documentation](https://cmake.org/cmake/help/latest/manual/cmake-presets.7.html)
+* [CMake Presets in Visual Studio](https://docs.microsoft.com/en-us/cpp/build/cmake-presets-vs?view=msvc-170)
+
+### Building from Command Line
```
Comment (nightcrawlerstudios) [inline readme.md:67]: I would move "Building from Command Line" to underneath "Building in Visual Studio" - the Visual studio instructions are simpler, and its also more common for windows devs to use VS. It provides a nice break in the detail and a good foundation for moving on with the rest of the readme

**Review by jaremieromer — APPROVED** [review]

**Pull request merged**